src/components/hand-controls.js

```diff
@@ -74,34 +126,116 @@ module.exports.Component = registerComponent('hand-controls', {
    * @param {string} evt the event associated to the button
    */
   handleButton: function (button, evt) {
-    var el = this.el;
     var isPressed = evt === 'down';
+    var isTouched = evt === 'touchstart';
+    var shouldAnimate = true;
     switch (button) {
       case 'trackpad':
-        if (isPressed === this.trackpadPressed) { return; }
-        this.trackpadPressed = isPressed;
-        this.playAnimation('thumb', !isPressed);
-        evt = isPressed ? 'thumbup' : 'thumbdown';
-        el.emit(evt);
+        if (evt.indexOf('touch') === 0) {
+          if (isTouched === this.trackpadTouched) { return; }
+          this.trackpadTouched = isTouched;
+        } else {
+          if (isPressed === this.trackpadPressed) { return; }
+          this.trackpadPressed = isPressed;
+        }
         break;
       case 'trigger':
-        if (isPressed === this.triggerPressed) { return; }
-        this.triggerPressed = isPressed;
-        this.playAnimation('pointing', !isPressed);
-        evt = isPressed ? 'pointup' : 'pointdown';
-        el.emit(evt);
+        if (evt.indexOf('touch') === 0) {
+          if (isTouched === this.triggerTouched) { return; }
+          this.triggerTouched = isTouched;
+        } else {
+          if (isPressed === this.triggerPressed) { return; }
+          this.triggerPressed = isPressed;
+        }
         break;
       case 'grip':
-        if (isPressed === this.gripPressed) { return; }
-        this.gripPressed = isPressed;
-        this.playAnimation('close', !isPressed);
-        evt = isPressed ? 'gripclose' : 'gripopen';
-        el.emit(evt);
+        if (evt.indexOf('touch') === 0) {
+          if (isTouched === this.gripTouched) { return; }
+          this.gripTouched = isTouched;
+        } else {
+          if (isPressed === this.gripPressed) { return; }
+          this.gripPressed = isPressed;
+        }
+        break;
+      case 'thumbstick':
+        if (isPressed === this.thumbstickPressed) { return; }
+        this.thumbstickPressed = isPressed;
+        shouldAnimate = false;
+        break;
+      case 'AorX':
+        if (isTouched === this.AorXTouched) { return; }
+        this.AorXTouched = isTouched;
+        break;
+      case 'menu':
+        if (isTouched === this.menuTouched) { return; }
+        this.menuTouched = isTouched;
+        break;
+      case 'surface':
+        if (isTouched === this.surfaceTouched) { return; }
+        this.surfaceTouched = isTouched;
         break;
     }
+    if (shouldAnimate) { this.animate(); }
   },
 
-  /**
+  animate: function () {
+    if (this.gripPressed) {
+      if (this.surfacePressed || this.surfaceTouched ||
+          this.menuTouched || this.AorXTouched ||
+          this.trackpadPressed || this.trackpadTouched) {
+        if (!this.triggerPressed && !this.triggerTouched) {
+          // point
+          this.playAnimation('pointing', false);
```

Comment:
Can we just create a variable `var animation` that we asign in each of the conditional branches? At the end of the function we call `this.playAnimation` once. It makes debugging easier.